gtdynamics/utils/values.h

```diff
@@ -39,45 +39,60 @@ class KeyDoesNotExist : public gtsam::ValuesKeyDoesNotExist {
   Key definitions.
  ************************************************************************* */
 /// Shorthand for q_j_t, for j-th joint angle at time t.
-inline DynamicsSymbol JointAngleKey(int j, int t = 0) {
+inline gtsam::Key JointAngleKey(int j, int t = 0) {
   return DynamicsSymbol::JointSymbol("q", j, t);
 }
 
 /// Shorthand for v_j_t, for j-th joint velocity at time t.
-inline DynamicsSymbol JointVelKey(int j, int t = 0) {
+inline gtsam::Key JointVelKey(int j, int t = 0) {
   return DynamicsSymbol::JointSymbol("v", j, t);
 }
 
 /// Shorthand for a_j_t, for j-th joint acceleration at time t.
-inline DynamicsSymbol JointAccelKey(int j, int t = 0) {
+inline gtsam::Key JointAccelKey(int j, int t = 0) {
   return DynamicsSymbol::JointSymbol("a", j, t);
 }
 
 /// Shorthand for T_j_t, for torque on the j-th joint at time t.
-inline DynamicsSymbol TorqueKey(int j, int t = 0) {
+inline gtsam::Key TorqueKey(int j, int t = 0) {
   return DynamicsSymbol::JointSymbol("T", j, t);
 }
 
 /// Shorthand for p_i_t, for COM pose on the i-th link at time t.
-inline DynamicsSymbol PoseKey(int i, int t = 0) {
+inline gtsam::Key PoseKey(int i, int t = 0) {
   return DynamicsSymbol::LinkSymbol("p", i, t);
 }
 
 /// Shorthand for V_i_t, for 6D link twist vector on the i-th link.
-inline DynamicsSymbol TwistKey(int i, int t = 0) {
+inline gtsam::Key TwistKey(int i, int t = 0) {
   return DynamicsSymbol::LinkSymbol("V", i, t);
 }
 
 /// Shorthand for A_i_t, for twist accelerations on the i-th link at time t.
-inline DynamicsSymbol TwistAccelKey(int i, int t = 0) {
+inline gtsam::Key TwistAccelKey(int i, int t = 0) {
   return DynamicsSymbol::LinkSymbol("A", i, t);
 }
 
 /// Shorthand for F_i_j_t, wrenches at j-th joint on the i-th link at time t.
-inline DynamicsSymbol WrenchKey(int i, int j, int t = 0) {
+inline gtsam::Key WrenchKey(int i, int j, int t = 0) {
   return DynamicsSymbol::LinkJointSymbol("F", i, j, t);
 }
 
+/// Shorthand for C_i_c_k, for contact wrench c on i-th link at time step k.
+inline gtsam::Key ContactWrenchKey(int i, int c, int k = 0) {
+  return DynamicsSymbol::LinkJointSymbol("C", i, c, k);
+}
+
+/* Shorthand for dt_k, for duration for timestep dt_k during phase k. */
+inline gtsam::Key PhaseKey(int k) {
+  return DynamicsSymbol::SimpleSymbol("dt", k);
+}
+
+/* Shorthand for t_k, time at time step k. */
+inline gtsam::Key TimeKey(int k) {
+  return DynamicsSymbol::SimpleSymbol("t", k);
+}
+
```

Comment:
I didn't notice yetong's key functions changing, did you do those too?  e.g. `Actuator.MassKey`, ...

not sure where those would be...

Reply:
Those are in `jumping_robot.py` and already call `.key` on a Symbol object, so that looks good.

Reply:
That shall not be a problem